src/helpers/figures/charts/chart_common.ts

```diff
@@ -258,6 +253,75 @@ export function toExcelDataset(getters: CoreGetters, ds: DataSet): ExcelChartDat
   };
 }
 
+export function toExcelLabelRange(
+  getters: CoreGetters,
+  labelRange: Range | undefined,
+  dataSetsHaveTitle?: boolean
+) {
+  if (!labelRange) return undefined;
+  let zone = {
+    ...labelRange.zone,
+  };
+  if (dataSetsHaveTitle) {
+    if (labelRange.zone.bottom > labelRange.zone.top) {
+      zone.top = zone.top + 1;
+    } else {
+      zone.left = zone.left + 1;
+    }
+  }
+  const range = labelRange.clone({ zone });
+  return getters.getRangeString(range, "forceSheetReference");
+}
+
+export function convertExcelDatasetToSheetXC(
+  ds: ExcelChartDataset,
+  dataSetsHaveTitle: boolean
+): string {
+  let { sheetName, xc } = splitReference(ds.range);
+  if (sheetName) {
+    sheetName = sheetName + "!";
```

Comment:
this causes an issue, we have to discuss this tomorrow. THe sheetName yields does not have quotes its the pure sheet name. So when you combine your sheetName and dataXC later on, you should check taht hte quotes are present if  necessary. a part of this logic is present and is inadequately called `getComposerSheetName` so you can use  this ATM but it highlights the need to set some kind of guidelines on how to process sheetNames